Dockerfile

```diff
@@ -1,7 +1,28 @@
-FROM ubuntu:latest
+# CHIA BUILD STEP
+FROM python:3.9 AS chia_build
 
-EXPOSE 8555
-EXPOSE 8444
+RUN dpkg-reconfigure -f noninteractive tzdata
+
+ARG BRANCH=latest
```

Comment:
If we are restructuring, it's probably a good idea to also move this ARG to right before the first use of the argument. That is: down to right before the `RUN echo "cloning ${BRANCH}" ...`.

ARG creates a separate image layer. If the ARG is as late as possible, all layers before the ARG can be re-used even if the build argument changes. In practice, for this Dockerfile it would mean that even if we rebuild for a different BRANCH, the the apt update/apt install layer of a previous build can be reused.

(NB: I'm not affiliated with Chia Inc and I cannot merge, so this suggestion is no "official" remark.)